esmf_regrid/tests/unit/schemes/test_ESMFAreaWeighted.py

```diff
@@ -62,3 +63,10 @@ def test_invalid_tgt_location():
     match = "For area weighted regridding, target location must be 'face'."
     with pytest.raises(ValueError, match=match):
         _ = ESMFAreaWeighted(tgt_location="node")
+
+
+def test_non_degree_crs():
+    """Test for coordinates with non-degree units."""
+    expected_sum = 50.86147272655136
+    expected_unmasked = 21
+    _test_non_degree_crs(ESMFAreaWeighted, expected_sum, expected_unmasked)
```

Comment:
This is certainly a difficult one to get right, but to make the code more understandable I think both the setup and expected results need to be in the same place. So either:

- Contain the expected results within `_test_non_degree_crs()`, obviously as some sort of tuple given the various schemes.
- Generate the source and target in the function here, and pass those down to `_test_non_degree_crs()` as more arguments.